src/new-client/src/services/currencyPairs/currencyPairs.test.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useEffect, useRef } from "react"
+import { Subscribe } from "@react-rxjs/core"
+import { whenStream, reset } from "utils/mockClient"
+import { BehaviorSubject, defer, Observable, Subject } from "rxjs"
+import {
+  render,
+  screen,
+  waitFor,
+  act as componentAct,
+} from "@testing-library/react"
+import { renderHook, act as hookAct } from "@testing-library/react-hooks"
+
+import { CurrencyPair, CurrencyRaw, RawCurrencyPairUpdates } from "./types"
+
+import {
+  useCurrencyPairs,
+  currencyPairs$,
+  useCurrencyPair,
+  getCurrencyPair$,
+} from "./currencyPairs"
+import { UpdateType } from "services/utils"
+
+enum MockSymbols {
+  EURUSD = "EURUSD",
+  GBPJPY = "GBPJPY",
+}
+
+const mockCurrencyPairsRaw: Record<MockSymbols, CurrencyRaw> = {
+  [MockSymbols.EURUSD]: {
+    Symbol: "EURUSD",
+    RatePrecision: 5,
+    PipsPosition: 4,
+  },
+  [MockSymbols.GBPJPY]: {
+    Symbol: "GBPJPY",
+    RatePrecision: 1,
+    PipsPosition: 1,
+  },
+}
+
+const mockCurrencyPairs: Record<MockSymbols, CurrencyPair> = {
+  [MockSymbols.EURUSD]: {
+    symbol: "EURUSD",
+    ratePrecision: 5,
+    pipsPosition: 4,
+    base: "EUR",
+    terms: "USD",
+  },
+  [MockSymbols.GBPJPY]: {
+    symbol: "GBPJPY",
+    ratePrecision: 1,
+    pipsPosition: 1,
+    base: "GBP",
+    terms: "JPY",
+  },
+}
+
+const useCurrencyPairWithUpdates = (symbol: MockSymbols) => {
```

Comment:
Do we want to use this as a pattern in all our tests and create a utility that accepts a hook and returns the hook with an update-count closure?

Reply:
maybe? There are lots of things that could be "abstracted" and yes, this could be one of those... If and when we start using it in other tests we can create that abstraction, for sure.